proxy/src/main/java/com/velocitypowered/proxy/connection/client/ClientPlaySessionHandler.java

```diff
@@ -277,7 +277,11 @@ public boolean handle(TabCompleteRequest packet) {
 
   @Override
   public boolean handle(PluginMessage packet) {
-    VelocityServerConnection serverConn = player.getConnectedServer();
+    // Handling edgecase when packet with FML client handshake (state COMPLETE)
+    // arrives after JoinGame packet from destination server
+    VelocityServerConnection serverConn = player.getConnectedServer() != null
+            ? player.getConnectedServer() : player.getConnectionInFlight();
```

Comment:
I'm not a big fan of this approach. We should limit this to if we're handling a Forge message specifically, or we're handling a Forge client (as server switch failures with Forge are considered fatal once we send the reset packet to the client).